### ISSUE TITLE
Preservation of story nuggets

### ISSUE THREAD

**Issue by shentino** [ChatTheatre/SkotOS#51]
I would like to lobby for the preservation of storynuggets as a character generation interface, I think it was used on marrach if I'm not mistaken unless I misread something, and if they still work I don't think they should be removed.

Though by contrast I agree bilbo needs to go.

**Comment by noahgibbs** [comment]
Fair enough. I think we have some time on this one - I hadn't been planning to remove them immediately. I don't know enough to have an opinion here. I think @ChristopherA would be the one to convince.

**Comment by shentino** [comment]
You did however commit them to the deprecation list which is how I found out about this.

Thing is if they still work I'd like them kept around as a second option.  If anything I think they just need to be better documented.

**Comment by noahgibbs** [comment]
At the moment I'm taking documentation from an external site, written by other folks, and committing it where I think it goes. Similarly, the information on Regency clothing that I dropped into the gables_game repo wasn't written by me -- I'm just putting stuff that currently isn't in the repos, into the repos.

**Comment by noahgibbs** [comment]
My hope is that a lot of the disparate external sources (e.g. the Builders section of the TWiki) can be merged into a central location, which will let us figure out what's correct, what's out of date and what can just go in terms of documentation. The deprecation list came from HackMD.

**Comment by ChristopherA** [comment]
The problem is that StoryNuggets doesn’t work in stock Gables. There is some specific code in Marrach that must not exist in our codebase.

If you want to write a new SkotOS plugin to make that feature available in the future, that would be great. 

However anything that is not functional should be removed from base SkotOS. In addition almost  everything not necessary for almost all games or a makes for great demo should be converted to optional plugins.

Either way StoryNuggets needs to be deprecated from base SkotOS. We have backups in our repos now should someone want to try to make it work in the future.

**Comment by shentino** [comment]
That's surprising to me ChrisA, I was under the impression that StoryNuggets worked just fine in that character creation page that let you redo your attributes, unless I'm horribly confused.

Is it possible that it was one of the things that got broken by the cold boot process and only worked during a live instance of Gables?

**Comment by noahgibbs** [comment]
Any idea if it works on the current gables.chattheatre.com? That isn't the cold-boot code (yet, we'll get there). Though it's pretty different code in general.

**Comment by shentino** [comment]
I just remember using the character creation page to reset my attributes (eye color, hair, nose, etc) and I was sure that it used StoryNuggets

**Comment by noahgibbs** [comment]
It *does* look like the current character editor uses story nuggets in some capacity: https://github.com/ChatTheatre/SkotOS/blob/master/skoot/data/vault/Theatre/Theatres/Tavern.xml

**Comment by ChristopherA** [comment]
We should ask @saraht45 — I don’t believe it works with current start story.

**Comment by saraht45** [comment]
I personally don't use story nuggets as I feel it is too limited. The current Gables start story that I wrote does not use it, but that's not to say there aren't other references to it floating around. I know there is an "edit character" page but I think the idea is to use +traits/@traits to modify chatter bodies. My only concern would be if it's hardwired into SkotOS then removing it might break something on a lower level.

My vote is to get rid of it. As Chris said, it wasn't working right on Gables anyway.

**Comment by noahgibbs** [comment]
If Sarah says it wasn't working right on Gables, that means it's also not working right on prod Gables - so that means she's not referring to a cold-boot or CTSkotOS problem, since that's not the codebase she's using.

**Comment by shentino** [comment]
> It _does_ look like the current character editor uses story nuggets in some capacity: https://github.com/ChatTheatre/SkotOS/blob/master/skoot/data/vault/Theatre/Theatres/Tavern.xml

If I'm not mistaken this XML dump cites storynuggets as a component of the theatre object if the tags are anything to go by, since the tag types reference the SIDs being used to parse the xml.

**Comment by shentino** [comment]
So for the moment this does imply that existing code would indeed be broken if we got rid of it.  I'm not really clean to get delete happy with the code unless it's broken in a way that can't be avoided.

**Comment by noahgibbs** [comment]
You're correct -- if we were going to remove story nuggets, that would require first changing the start story in CTSkotOS, which uses them.

**Comment by shentino** [comment]
I also would opine that theatres are important in their own right for thematic reasons.

I'm just not really feeling comfortable just gutting stuff, especially if things change later that make it hard to reintegrate it later once it's fixed.

**Comment by ChristopherA** [comment]
There is nothing gutted. It all exists in GitHub archival repos now, and can be easily brought back into a SkotOS instance if needed. But it if isn't needed for root SkotOS repo, it shouldn't be there — it should either be in its own archival repo, or if important enough, in a repo that supports @noahgibbs new plugin architecture so that it can be easily added to a SkotOS instance.
